src/backend/distributed/transaction/backend_data.c

```diff
@@ -401,27 +401,39 @@ BackendManagementShmemSize(void)
 
 
 /*
- *  InitializeBackendData is called per backend and does the
- *  required initialization.
+ * InitializeBackendData initialises MyBackendData to the shared memory segment
+ * belonging to the current backend.
+ *
+ * The function is called through CitusHasBeenLoaded when we first detect that
+ * the Citus extension is present, and after any subsequent invalidation of
+ * the metadata.
```

Comment:
So, `invalidation of the metadata.` seems not accurate? That could happen on `drop extension` - `alter pg_dist_partition` or `vacuum pg_dist_partition` or any others?